tests/python/pants_test/backend/python/test_interpreter_cache.py

```diff
@@ -171,3 +173,32 @@ def test_setup_cached_warm(self):
   def test_setup_cached_cold(self):
     with self._setup_cache() as (cache, _):
       self.assertEqual([], list(cache._setup_cached()))
+
+  def test_interpreter_from_relpath_purges_stale_interpreter(self):
+    """
+    Simulates a stale interpreter cache and tests that _interpreter_from_relpath
+    properly detects it and removes the stale dist directory.
+
+    See https://github.com/pantsbuild/pants/issues/3416 for more info.
+    """
+    with temporary_dir() as temp_dir:
+      # Setup a interpreter distribution that we can safely mutate.
+      test_interpreter_binary = os.path.join(temp_dir, 'python2.7')
+      src = '/usr/bin/python2.7'
```

Comment:
It realy is more robust to `os.path.realpath(sys.executable)` - you know that oath must exist and points to the actual binary file and not a symlink.

Reply:
I tried to use the output of `which python2.7` like above tests with py27 path, however only copying the binary for that path gave me issues with linking against the python runtime. I'll give this a shot.

Reply:
I am experiencing a different issue with sys.exe: 
```[tw-mbp-clivingston pants (clivingston/PY-135)]$ /var/folders/7n/x4k4tqh94k9gwm61g046m2340000gn/T/tmp9p8l3H/python2.7
Could not find platform independent libraries <prefix>
Could not find platform dependent libraries <exec_prefix>
Consider setting $PYTHONHOME to <prefix>[:<exec_prefix>]
ImportError: No module named site```
(due to the venv I think)

I think that /usr/bin/python2.7 gets me PATH coherence for free which is why I did that. I'm happy to play around with PYTHONPATH or temp PATH setting.